webapp/src/components/BuyParcelByMortgagePage/BuyParcelByMortgagePage.container.js

```diff
@@ -2,26 +2,39 @@ import { withRouter } from 'react-router'
 import { connect } from 'react-redux'
 import { push } from 'react-router-redux'
 
-import BuyParcelByMortgagePage from './BuyParcelByMortgagePage'
-import { createMortgageRequest } from 'modules/mortgage/actions'
 import { locations } from 'locations'
 import { getMatchParamsCoordinates } from 'modules/location/selectors'
-import { isRequestingMortgageTransactionIdle } from 'modules/mortgage/selectors'
-import { getPublicationByCoordinate } from 'modules/publication/selectors'
+import { createMortgageRequest } from 'modules/mortgage/actions'
+import {
+  isRequestingMortgageTransactionIdle,
+  getError as getMortgageError
+} from 'modules/mortgage/selectors'
 import { getWallet, isConnected, isConnecting } from 'modules/wallet/selectors'
-import { getError as getMortgageError } from 'modules/mortgage/selectors'
+import {
+  getData as getAuthorizations,
+  isLoading
+} from 'modules/authorization/selectors'
+import { getPublicationByCoordinate } from 'modules/publication/selectors'
+import BuyParcelByMortgagePage from './BuyParcelByMortgagePage'
 
 const mapState = (state, ownProps) => {
   const { x, y } = getMatchParamsCoordinates(ownProps)
+  const wallet = getWallet(state)
+
+  let authorization
+
+  if (wallet) {
+    authorization = getAuthorizations(state)[wallet.address]
+  }
```

Comment:
this pattern:

```javascript
  let authorization
   if (wallet) {
    authorization = getAuthorizations(state)[wallet.address]
  }
```

repeats on every container that needs `authorization`. Do you thinks is better to do something like `const authorizations = getAuthorizations(state)[getAddress(state)]` ?

Reply:
What do you think of using a HOC which handles authorizations for children components?

```
class WithAuthorization extends React.Component {}
withAuthorization(connect(toState, toDispatch)(BuyParcelByMortgagePage))
```

Maybe we can do it in another PR

Reply:
I like that! I should put more thought into it but I like that it's composable and easily removable, but yes we can leave it for another PR